### PR TITLE
Some test code clean-up

### PR DESCRIPTION
**Pull request by offa**
Some punctual clean-up in the jenkins-test module.

- Anonymous classes replaced by lambda
- Declared exceptions removed, which are already included through a more general class
- Commented out debug code removed
- Minor code tweaks

### Proposed changelog entries

_None_

### Proposed upgrade guidelines

_None_

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — APPROVED** [review]
The build failed due to some unused imports. You can fix those automatically by running `mvn spotless:apply`. One tip for future reference is to run `mvn verify -DskipTests` locally prior to posting a PR. That runs the build and static analysis checks (which take about a minute) but skips the tests (which take several hours).

Pending a passing build this PR looks good to me.

**Review by timja — APPROVED** [review]

**Comment by offa** [comment]
Good to know, thanks! And thanks @timja for the fixup.

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
